src/js/components/ToggleCheckboxList/ToggleCheckboxList.tsx

```diff
@@ -35,12 +36,31 @@ const ToggleCheckboxList: React.FunctionComponent<Props> = ({
   activeRows,
   headerStatus,
   setActiveRows,
-  setHeaderStatus
+  setHeaderStatus,
+  sortOrder = 'asc'
 }: Props) => {
   // Dynamically create refs outside of the mapping below for performance reasons
   const bodyRefs = React.useRef([]);
   bodyRefs.current = list.map((_, i) => bodyRefs.current[i] ?? React.createRef());
 
+  const sort = (list: ExpandableList[]) => {
+    if (sortOrder === 'asc') {
+      // Sort headers
+      list.sort((prev: ExpandableList, curr: ExpandableList) => prev.heading[0] > curr.heading[0] ? 1 : 0);
+      // Sort sub-rows
+      list.map(row => row.rows.sort((a, b) => a[0] > b[0] ? 1 : 0));
+    }
+
+    if(sortOrder === 'desc') {
+      // Sort headers
+      list.sort((prev: ExpandableList, curr: ExpandableList) => prev.heading[0] < curr.heading[0] ? 1 : 0);
+      // Sort sub-rows
+      list.map(row => row.rows.sort((a, b) => a[0] < b[0] ? 1 : 0));
+    }
+  }
+
+  sort(list);
```

Comment:
This should probably be in a use effect. Otherwise, this will sort on every render which is probably way overkill.